opafiber/fiber.go

```diff
@@ -9,66 +9,62 @@ import (
 	"io"
 )
 
+type InputCreationFunc func(c *fiber.Ctx) (map[string]interface{}, error)
+
 type Config struct {
 	RegoPolicy            io.Reader
 	RegoQuery             string
 	IncludeHeaders        []string
 	IncludeQueryString    bool
 	DeniedStatusCode      int
 	DeniedResponseMessage string
+	InputCreationMethod   InputCreationFunc
 }
 
 func New(cfg Config) fiber.Handler {
-	if cfg.RegoQuery == "" {
-		panic(fmt.Sprint("rego query can not be empty"))
-	}
-
-	if cfg.DeniedStatusCode == 0 {
-		cfg.DeniedStatusCode = fiber.StatusBadRequest
+	err := cfg.fillAndValidate()
+	if err != nil {
+		panic(err)
 	}
-
 	readedBytes, err := io.ReadAll(cfg.RegoPolicy)
 	if err != nil {
 		panic(fmt.Sprint("could not read rego policy %w", err))
 	}
-
 	query, err := rego.New(
 		rego.Query(cfg.RegoQuery),
 		rego.Module("policy.rego", utils.UnsafeString(readedBytes)),
 	).PrepareForEval(context.Background())
-
 	if err != nil {
 		panic(fmt.Sprint("rego policy error: %w", err))
 	}
-
 	return func(c *fiber.Ctx) error {
-		input := map[string]interface{}{
-			"method": c.Method(),
-			"path":   c.Path(),
+		input := make(map[string]interface{})
+		input, err = cfg.InputCreationMethod(c)
```

Comment:
No need for `input := make(map[string]interface{})` line.

`input, err := cfg.InputCreationMethod(c)` already covers it.